src/components/StudyView/StudyView.tsx

```diff
@@ -51,29 +50,45 @@ const StudyView: React.FC<StudyViewProps> = ({
   const templateVisibleSlots = useMemo(() => {
     const fieldVisibilityInfo: TemplateVisibleSlots[] = [];
     if (submission.data) {
-      const packageName = submission.data.metadata_submission.packageName;
-      if (packageName !== "") {
-        const template = TEMPLATES[packageName];
+      if (!("field_notes_metadata" in submission.data)) {
+        return undefined;
+      }
+      const templates = getSubmissionTemplates(submission.data);
+      templates.forEach((templateName) => {
+        const template = TEMPLATES[templateName];
         fieldVisibilityInfo.push({
-          template: packageName,
+          template: templateName,
           templateDisplay: template.displayName,
           visibleSlots:
             submission.data.field_notes_metadata?.fieldVisibility?.[
-              packageName
+              templateName
             ],
         });
-      }
+      });
     }
     return fieldVisibilityInfo;
   }, [submission.data]);
 
-  useIonViewDidEnter(() => {
-    if (openSlotSelectorModalOnEnter) {
-      setModalTemplateVisibleSlots(templateVisibleSlots[0]);
+  useEffect(() => {
+    // If the slot selector is not already open, iterate through the study's templates. If one of
+    // them has no visible slot information (this could be because it is a brand-new study or
```

Comment:
I felt confused when I first read this comment (reading it as "If the slot information of one of the templates is not visible..."). I find this alternative to be more clear:

```suggestion
    // them has no slot visibility information (this could be because it is a brand-new study or
```